packages/apollo/src/commands/service/__tests__/list.test.ts

```diff
@@ -0,0 +1,229 @@
+                graphVariant: "current",
+                updatedAt: "2019-05-13T14:56:04.172Z",
+                __typename: "FederatedImplementingService",
+                name: "accounts",
+                url: "http://localhost:4001/graphql"
+              },
+              {
+                graphID: "maya-federation-demo",
+                graphVariant: "current",
+                updatedAt: "2019-05-16T19:21:10.243Z",
+                __typename: "FederatedImplementingService",
+                name: "products",
+                url: "http://localhost:4003/graphql"
+              }
+            ],
+            __typename: "FederatedImplementingServices"
+          }
+        }
+      }
+    });
+}
+
+/**
+ * Mock network requests for a successful service list of a nonfederated graph.
+ */
+function mockServiceListNonFederated() {
+  nock(stagingAPI, {
+    encodedQueryParams: true
+  })
+    .post(
+      "/api/graphql",
+      ({ operationName }) => operationName === "ListServices"
+    )
+    .reply(200, {
+      data: {
+        service: {
+          implementingServices: {
+            services: [],
+            __typename: "FederatedImplementingServices"
+          }
+        }
+      }
+    });
+}
+
+describe("service:list", () => {
+  let originalChalkEnabled;
+
+  beforeEach(() => {
+    originalChalkEnabled = chalk.enabled;
+    chalk.enabled = false;
+
+    // Clean console log capturing before tests in the event that `afterEach` was not run successfully.
+    uncaptureApplicationOutput();
+
+    // Clean up all network mocks before tests in the event that `afterEach` was not run successfully.
+    nock.cleanAll();
+
+    nock.disableNetConnect();
+
+    // Set the jest timeout to be longer than the default 5000ms to compensate for slow CI.
+    jest.setTimeout(15000);
+  });
+
+  afterEach(() => {
+    chalk.enabled = originalChalkEnabled;
+
+    // Clean up console log mocking
+    uncaptureApplicationOutput();
+
+    // Clean up all network mocks and restore original functionality
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+  describe("integration", () => {
+    describe("should list services correctly for a federated service", () => {
+      it("vanilla", async () => {
+        captureApplicationOutput();
+        mockServiceListFederated();
+
+        expect.assertions(2);
+
+        await expect(
+          ServiceList.run([...cliKeyParameter])
+        ).resolves.not.toThrow();
+
+        // Inline snapshots don't work here due to https://github.com/facebook/jest/issues/6744.
+        expect(uncaptureApplicationOutput()).toMatchSnapshot();
+      });
+    });
+
+    describe("should display non federated message for a non federated service", () => {
+      it("vanilla", async () => {
+        captureApplicationOutput();
+        mockServiceListNonFederated();
+
+        expect.assertions(2);
+
+        await expect(
+          ServiceList.run([...cliKeyParameter])
```

Comment:
Same here

Reply:
Consider a rename, since `cliKeyParameter` is itself an array. Maybe `cliOptions` or similar that represents a list of things we're passing.